### PR TITLE
Start working on nuget templates.

### PR DESCRIPTION
**Pull request by jtkech**
So, i created a project which is part of the solution and on packaging it embeds in a content folder our 3 templates which are not part of the solution and with their `template.json` configuration file.

This is in the `.csproj` that we specify the msbuild nuspec property `PackageType = Template`. So we don't need to have a `.nuspec` file, it is generated when packaging this project through VS or dotnet.

**TODO**: Try to install the generated package and use its templates through dotnet new.

As a reminder, for all our generated packages maybe good to add some msbuild nuspec properties e.g in in our `Commons.props` so that they appear in the embedded .nuspec files. E.g `authors` and `owners`.

### CODE REVIEW

File: src/Templates/OrchardCore.Cms.Templates/content/OrchardCore.Templates.Module/OrchardCore.Templates.Module.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OrchardCore.Module.Targets" Version="1.0.0-beta2-6657" />
```
Comment (sebastienros) [inline src/Templates/OrchardCore.Cms.Templates/content/OrchardCore.Templates.Module/OrchardCore.Templates.Module.csproj]: The version need to come from the `$(BuilderVersion)` that is set on the CI (it's also set to 1.0 locally)

File: src/Templates/OrchardCore.Cms.Templates/content/OrchardCore.Templates.Theme/OrchardCore.Templates.Theme.csproj [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk.Razor">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OrchardCore.Theme.Targets" Version="1.0.0-beta2-6657" />
```
Comment (sebastienros) [inline src/Templates/OrchardCore.Cms.Templates/content/OrchardCore.Templates.Theme/OrchardCore.Templates.Theme.csproj]: Same here

**Comment by sebastienros** [comment]
You can keep the hard coded versions if you want to try locally, but then use $(BuilderVersion) once the tests are done.

**Comment by jtkech** [comment]
It's new for me, just learnt it yesterday. Yes, the version was for testing that i didn't do yet.

- But to use `$(BuildVersion)`, when packaging the parent templates project, we would need a more complex script  to first update the child template project files embedded as contents.

- Maybe we could start by specifying release versions, and update the templates on each release version. Hmm, but in this case it would not make sense to pack the parent templates project each time.

**Testing**

- Just tried, it works for `octheme`, the project is correctly created. I will check it after under VS because right now it's not the main part.

- But it didn't find `ocmodule` and `occms`. So first i will try to fix this.

**Comment by Skrypt** [comment]
@sebastienros  Why are you removing the Nuget doc chapter? You will need to also remove the anchor links if you do that.

**Comment by Skrypt** [comment]
No need to initialize templates anymore?

**Comment by sebastienros** [comment]
@Skrypt the first command is to install the templates (what you call initialize I assume)

**Comment by xperiandri** [comment]
How to install 1.0.0-beta2 templates?
`dotnet new --install "OrchardCore.Templates::1.0.0-beta2"` does not work

**Pull request merged**